gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -81,26 +93,53 @@ public static void main(String[] argv) throws IOException {
-                "glob:**/data/**/*.gz");
-        final Path gigawordDir = parameters.getExistingDirectory(PARAM_GIGAWORD_DIRECTORY_PATH).toPath();
-        try (Stream<Path> gigawordFiles = Files.walk(gigawordDir)) {
-          gigawordFiles
-                  .filter(gzippedConcatenatedFilePattern::matches)
-                  .forEach(gzippedConcatenatedFile -> {
-                    try {
-                          index(client, gzippedConcatenatedFile, indexName);
-                    } catch (IOException e) {
-                      throw new RuntimeException(e);
-                    }
-                  });
-        }
+      final String format = parameters.getString(PARAM_FORMAT);
+      final String lang = parameters.getOptionalString(PARAM_LANGUAGE).or("EN");
+      final double fractionDocAllowToFail =
+          Double.parseDouble(parameters.getOptionalString(PARAM_FRACTIOIN_DOCS_ALLOWED_TO_FAIL)
+              .or("0.0"));
+      final int sentenceLimit = parameters.getOptionalInteger(SENTENCE_LIMIT).or(100);
+      final Path corpusDirPath = parameters.getExistingDirectory(PARAM_CORPUS_DIRECTORY_PATH).toPath();
+      PathMatcher filePattern;
+      if (format.equalsIgnoreCase("LTF")) {
+        filePattern = FileSystems.getDefault().getPathMatcher("glob:**.ltf.zip");
       } else {
-        File concatenatedFileToIndex = parameters.getExistingFile(PARAM_GIGAWORD_FILEPATH);
-        index(client, concatenatedFileToIndex.toPath(), indexName);
+        filePattern = FileSystems.getDefault().getPathMatcher("glob:**/data/**/*.gz");
+      }
+
+      try (Stream<Path> corpusFiles = Files.walk(corpusDirPath)) {
+        corpusFiles
+            .filter(filePattern::matches)
+            .forEach(
+                gzippedConcatenatedFile -> {
+                  try {
+                    // If file ends with .xml.gz, this may be the wrong version of Gigaword.
+                    if ((format.equalsIgnoreCase("gigaword"))
+                        && gzippedConcatenatedFile.toString().toLowerCase().endsWith(".xml.gz")) {
+                      log.warn("Indexing file ending with .xml.gz. This may indicate incorrect Gigaword version. "
+                          + "Make sure you are using the plain version LDC2011T07.");
+                    }
+
+                    // we batch the documents in groups of 100 so we can get the efficiency gains from batching without
+                    // making huge requests of unbounded size
+                    Iterable<List<Article>> iterator;
+                    if (format.equalsIgnoreCase("ltf")) {
+                      try (LTFDocuments ltfDocuments = LTFDocuments.fromLTFZippedFile(gzippedConcatenatedFile)) {
+                        iterator = partition(ltfDocuments, BATCH_SIZE);
+                        index(client, iterator, indexName, lang, fractionDocAllowToFail, sentenceLimit);
+                      }
+                    } else {
```

Comment:
Explicitly check for `gigaword`. What if someone put `format: foo`?  Throw an exception on an unknown type.

Reply:
Fixed.